sendrecv/gst-java/src/main/java/WebrtcSendRecv.java

```diff
@@ -0,0 +1,253 @@
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.asynchttpclient.ws.WebSocket;
+import org.asynchttpclient.ws.WebSocketListener;
+import org.asynchttpclient.ws.WebSocketUpgradeHandler;
+import org.freedesktop.gstreamer.*;
+import org.freedesktop.gstreamer.Element.PAD_ADDED;
+import org.freedesktop.gstreamer.elements.DecodeBin;
+import org.freedesktop.gstreamer.elements.WebRTCBin;
+import org.freedesktop.gstreamer.elements.WebRTCBin.CREATE_OFFER;
+import org.freedesktop.gstreamer.elements.WebRTCBin.ON_ICE_CANDIDATE;
+import org.freedesktop.gstreamer.elements.WebRTCBin.ON_NEGOTIATION_NEEDED;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+import static org.asynchttpclient.Dsl.asyncHttpClient;
+
+/**
+ * Demo gstreamer app for negotiating and streaming a sendrecv webrtc stream
+ * with a browser JS app.
+ *
+ * @author stevevangasse
+ */
+public class WebrtcSendRecv {
+
+    private static final Logger logger = LoggerFactory.getLogger(WebrtcSendRecv.class);
+    private static final String REMOTE_SERVER_URL = "wss://webrtc.nirbheek.in:8443";
+    private static final String VIDEO_BIN_DESCRIPTION = "videotestsrc ! videoconvert ! queue ! vp8enc deadline=1 ! rtpvp8pay ! queue ! capsfilter caps=application/x-rtp,media=video,encoding-name=VP8,payload=97";
+    private static final String AUDIO_BIN_DESCRIPTION = "audiotestsrc ! audioconvert ! audioresample ! queue ! opusenc ! rtpopuspay ! queue ! capsfilter caps=application/x-rtp,media=audio,encoding-name=OPUS,payload=96";
+
+    private final String serverUrl;
+    private final String sessionId;
+    private final ObjectMapper mapper = new ObjectMapper();
+    private WebSocket websocket;
+    private WebRTCBin webRTCBin;
+    private Pipeline pipe;
+
+    public static void main(String[] args) throws Exception {
+        if (args.length == 0) {
+            logger.error("Please pass a session id from the signalling server: java -jar build/libs/gst-java.jar 1234");
+            return;
+        }
+        String sessionId = args[0];
+        String serverUrl = args.length == 2 ? args[1] : REMOTE_SERVER_URL;
```

Comment:
Can you ideally accept the server url in the same form as the other examples? (`--server SERVER`)

Reply:
No problem. Done

Reply:
awesome :)